lib/pin_code_text_field.dart

```diff
@@ -98,19 +107,25 @@ class PinCodeTextFieldState extends State<PinCodeTextField> {
   }
 
   void _onTextChanged(text) {
-    setState(() {
-      this.text = text;
-      if (text.length < currentIndex) {
-        strList[text.length] = "";
+    try {
+      setState(() {
+        this.text = text;
+        if (text.length < currentIndex) {
+          strList[text.length] = "";
+        } else {
+          strList[text.length - 1] =
+          widget.hideCharacter ? "\u25CF" : text[text.length - 1];
+        }
+        currentIndex = text.length;
+      });
+      if (text.length == widget.maxLength) {
+//      FocusScope.of(context).requestFocus(FocusNode());
+        widget.onDone(text);
       } else {
-        strList[text.length - 1] =
-            widget.hideCharacter ? "\u25CF" : text[text.length - 1];
+        widget.onDone("");
```

Comment:
`widget.onDone("")` is called when the text.length is not the maxLength. So it will get triggered every time a character is inserted. `onDone` should be called after all the details is filled.